IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -444,16 +438,55 @@ - (BOOL)saveAppMetadata:(nonnull MSIDAppMetadataCacheItem *)metadata
                                   error:error];
 }
 
-- (nullable MSIDAppMetadataCacheItem *)getAppMetadata:(nonnull MSIDAppMetadataCacheKey *)key
-                                              context:(nullable id<MSIDRequestContext>)context
-                                                error:(NSError * _Nullable * _Nullable)error
+- (BOOL)removeAppMetadata:(nonnull MSIDAppMetadataCacheItem *)appMetadata
+                  context:(nullable id<MSIDRequestContext>)context
+                    error:(NSError * _Nullable * _Nullable)error
 {
-    assert(key);
+    assert(appMetadata);
+    
+    MSID_LOG_VERBOSE(context, @"(Default cache) Removing app metadata with clientId %@, environment %@", appMetadata.clientId, appMetadata.environment);
+    
+    MSIDAppMetadataCacheKey *key = [[MSIDAppMetadataCacheKey alloc] initWithClientId:appMetadata.clientId
+                                                                         environment:appMetadata.environment
+                                                                            familyId:appMetadata.familyId
+                                                                         generalType:MSIDAppMetadataType];
+    
+    return [_dataSource removeItemsWithKey:key context:context error:error];
+}
+
+- (nullable NSArray<MSIDAppMetadataCacheItem *> *)getAppMetadataEntriesWithQuery:(nonnull MSIDAppMetadataCacheQuery *)cacheQuery
+                                                                         context:(nullable id<MSIDRequestContext>)context
+                                                                           error:(NSError * _Nullable * _Nullable)error
+{
+    assert(cacheQuery);
+    
+    MSID_LOG_VERBOSE_PII(context, @"(Default cache) Get app metadata entries with clientId %@, environment %@", cacheQuery.clientId, cacheQuery.environment);
     
-    MSID_LOG_VERBOSE(context, @"(Default cache) Get app metadata for key %@", key.logDescription);
-    MSID_LOG_VERBOSE_PII(context, @"(Default cache) Get app metadata for key %@", key.piiLogDescription);
+    NSArray<MSIDAppMetadataCacheItem *> *cacheItems = [_dataSource appMetadataEntriesWithKey:cacheQuery serializer:_serializer context:context error:error];
+    
+    if (!cacheQuery.exactMatch)
+    {
+        NSMutableArray<MSIDAppMetadataCacheItem *> *filteredResults = [NSMutableArray array];
+        
+        BOOL shouldMatchMetadata = !cacheQuery.clientId || !cacheQuery.environment;
```

Comment:
if it's not exact match, it means something needs to be matched. Since there's only clientId and environment, then this check and boolean are not necessary.

Reply:
As discussed offline, kept the check and fixed it to avoid filtering when not necessary.